src/contracts/pods/EigenPodManager.sol

```diff
@@ -675,12 +666,7 @@ contract EigenPodManager is
             shareAmounts[0] = podOwnerShares[podOwner];
             IStrategy[] memory strategies = new IStrategy[](1);
             strategies[0] = beaconChainETHStrategy;
-            delegationManager.decreaseDelegatedShares(podOwner, strategies, shareAmounts);
-
-            // undelegate the pod owner, if possible
-            if (stakerCanUndelegate(podOwner)) {
-                _undelegate(podOwner);
-            }
+            delegationManager.decreaseDelegatedShares(podOwner, strategies, shareAmounts, true/*undelegateIfPossible*/);
```

Comment:
Weird looping behavior here. I think this can be simplified and maybe we should pass "false" in here.

Reply:
agreed, especially with the commit that combines the two undelegation functions (https://github.com/Layr-Labs/eigenlayer-contracts/pull/176/commits/c5b9f55f650cecb3f75beb0d2114a322f73f4b3c) this has some weird reentrant behavior. simple fix would probably be passing "false" here

Reply:
addressed as part of PR https://github.com/Layr-Labs/eigenlayer-contracts/pull/183 -- or see the merge commit https://github.com/Layr-Labs/eigenlayer-contracts/pull/176/commits/75e1c96437486846dce1b391fe59c71a88598113